packages/interface/src/components/preferences-modal/README.md

```diff
@@ -0,0 +1,136 @@
+# PreferencesModal components
+
+`PreferencesModal`, `PreferencesModalTabs` and `PreferencesModalSection` are a family of components that greatly simplify building a WordPress editor preferences modal.
+
+`PreferencesModal` is a basic modal component that implements `Modal` from the `@wordpress/components` package.
+
+`PreferenceModalTabs` should be used to implement the responsive tab based menu structure within the preferences modal. It implements a `TabPanel` from `@wordpress/components` for desktop viewports, and a `Navigation` menu for smaller viewports.
+
+`PreferenceModalSection` represents a particular section within a tab that groups settings.
+
+See also the `PreferencesModalToggle` and `PreferencesModalFeatureToggle` components in this package that can be used to implement individual preference toggles.
+
+## Example
+
+```jsx
+function MyPreferencesModal( { closeModal }) {
+	// ...
+
+	const tabs = useMemo(
+		() => [
+			{
+				name: 'general',
+				title: __( 'General' ),
+				content: (
+					<PreferencesModalSection
+						title={ __( 'Appearance' ) }
+						description={ __(
+							'Customize options related to the block editor interface and editing flow.'
+						) }
+					>
+						<PreferencesModalFeatureToggle
+							scope="namespace/editor-name"
+							feature="reducedUI"
+							label={ __( 'Reduce the interface' ) }
+							help={ __(
+								'Compacts options and outlines in the toolbar.'
+							) }
+						>
+						<PreferencesModalFeatureToggle
+							scope="namespace/editor-name"
+							feature="focusMode"
+							label={ __( 'Spotlight mode' ) }
+							help={ __(
+								'Highlights the current block and fades other content.'
+							) }
+						/>
+					</PreferencesModalSection>
+				),
+			},
+			{
+				name: 'panels',
+				title: __( 'Panels' ),
+				content: (
+					<PreferencesModalSection
+						title={ __( 'Document settings' ) }
+						description={ __( 'Choose what displays in the panel.' ) }
+					>
+						<PreferencesModalToggle
+							label={ __( 'Permalink' ) }
+							onChange={ onTogglePermalink }
+							isChecked={ isPermalinkEnabled }
+						/>
+					</PreferencesModalSection>
+				),
+			}
+		],
+		[]
+	);
+
+	return (
+		<PreferencesModal onRequestClose={ closeModal }>
+			<PreferencesModalTabs tabs={ tabs } />
```

Comment:
I would personally prefer to see several nested `PreferencesModalTab` components here rather than the array passed as a prop.

Reply:
I agree with that. It's a side effect of building on the TabPanel component, which works this way:
https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/tab-panel/README.md

I suppose that component could be reworked, deprecated and replaced, or the modal could use its own very similar system.

Weighing the effort vs reward, I'm not sure the array API is bad enough to warrant such a refactor.